src/lib/UXDController.ts

```diff
@@ -28,158 +31,181 @@ export class UXDController {
     provider,
     controllerAddress,
     uxdTokenAddress,
+    market
   }: {
     provider: providers.Provider;
     controllerAddress: string;
     uxdTokenAddress: string;
+    market: string;
   }) {
     this.provider = provider;
-
+    this.market = market;
     this.controllerContract = UXDController__factory.connect(controllerAddress, this.provider);
     this.uxdContract = ERC20__factory.connect(uxdTokenAddress, this.provider);
 
     this.registerEventListeners();
   }
 
-  public mint({
-    market,
-    collateral,
-    ethAmount,
-    slippage,
-    signer,
-  }: {
-    market: string;
-    collateral: string;
-    ethAmount: BigNumber;
-    slippage: BigNumber;
-    signer: Signer;
-  }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).mint(
-      market,
-      collateral,
-      ethAmount,
-      slippage
-    );
-  }
-
-  public redeem({
-    market,
-    collateral,
-    uxdAmount,
-    slippage,
-    signer,
-  }: {
-    market: string;
-    collateral: string;
-    uxdAmount: BigNumber;
-    slippage: BigNumber;
-    signer: Signer;
-  }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).redeem(
-      market,
-      collateral,
-      uxdAmount,
-      slippage
-    );
-  }
-
-  public mintWithEth({
-    market,
-    ethAmount,
+  public async mint({
+    amount,
     slippage,
     signer,
+    collateralToken,
   }: {
-    market: string;
-    ethAmount: BigNumber;
-    slippage: BigNumber;
+    amount: number;
+    slippage: number;
     signer: Signer;
+    collateralToken?: string;
   }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).mintWithEth(
-      market,
-      slippage,
-      {
-        value: ethAmount,
-      },
-    );
-  }
-
-  public redeemEth({
-    market,
-    uxdAmount,
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    const ethSlippage = ethers.utils.parseEther(slippage.toString());
+    if (arguments.length == 4) {
+      return await this.mintWithERC20(
```

Comment:
Syntax: If inside of an `if` you call `return`, then you don't need an `else` as you already leave the function.

```
if (arguments.length == 4) {
      return await this.mintWithERC20(...);
}

if (arguments.length == 3) {
     return await this.mintWithETH(...);
}

throw Error("Invalid number of arguments");
```

Reply:
Done.